STD-26 | TheoryBlock Repository

zefirlover opened the pull request.

zefirlover committed:
ITheoryBlockRepository was added

zefirlover committed:
TheoryBlockRepo was added
Trying to fix it now

zefirlover committed:
TheoryBlockRepository.cs was added
I`ll try to fix it now

zefirlover committed:
TheoryBlockRepository.cs done

WerewolfNet commented on server/StudySharp.DomainServices/Repositories/TheoryBlockRepository.cs:
```diff
@@ -0,0 +1,71 @@
+
+        public async Task<OperationResult<List<TheoryBlock>>> GetTheoryBlocksByCourseIdAsync(int courseId)
+        {
+            await context.Courses.FindAsync(courseId);
```
Зачем эта строчка?

WerewolfNet commented on server/StudySharp.DomainServices/Repositories/TheoryBlockRepository.cs:
```diff
@@ -0,0 +1,71 @@
+        {
+            await context.Courses.FindAsync(courseId);
+            var theoryBlocks = await context.TheoryBlocks.Where(_ => _.CourseId == courseId).ToListAsync();
+            if (theoryBlocks == null)
```
Вроде как .ToListAsync() гарантированно сворачивает в список, поэтому там не может быть null

zefirlover replied:
А якшо в списку буде null

WerewolfNet commented on server/StudySharp.DomainServices/Repositories/TheoryBlockRepository.cs:
```diff
@@ -0,0 +1,71 @@
+
+        public async Task<OperationResult<TheoryBlock>> UpdateTheoryBlockAsync(TheoryBlock theoryBlock)
+        {
+            context.Entry(theoryBlock).State = EntityState.Modified;
```
theoryBlock может быть null

WerewolfNet commented on server/StudySharp.DomainServices/Repositories/TheoryBlockRepository.cs:
```diff
@@ -0,0 +1,71 @@
+{
+    public sealed class TheoryBlockRepository : ITheoryBlockRepository
+    {
+        private StudySharpDbContext context;
```
1)Добавь readonly
2)context => _context

WerewolfNet commented on server/StudySharp.DomainServices/Repositories/TheoryBlockRepository.cs:
```diff
@@ -0,0 +1,71 @@
+    {
+        private StudySharpDbContext context;
+
+        public TheoryBlockRepository(StudySharpDbContext _context)
```
_context => context

WerewolfNet reviewed (commented):
Везде забыл добавить await _context.SaveChangesAsync();

zefirlover committed:
TheoryBlockRepository.cs redacted, pre-dev 1
_context is now readonly
if statement and FindAsync in GetTheoryBlocksByCourseIdAsync was deleted
"potentially deleted if statement" in Update was added

zefirlover committed:
Minor changes, pre-dev 2
"potentially deleted if statement" comment was removed

zefirlover committed:
SaveChangesAsync was added, pre-dev 3
"()" in return new OperationResult were removed too

zefirlover committed:
Method AlternativeResult was added to implement DRY principle, pre-dev 4

WerewolfNet commented on server/StudySharp.DomainServices/Repositories/TheoryBlockRepository.cs:
```diff
@@ -0,0 +1,74 @@
+            _context = context;
+        }
+
+        private OperationResult<TheoryBlock> AlternativeResult(string errorText)
```
Приватные методы всегда идут после публичных

WerewolfNet replied:
Не уверен по поводу функционала в самом методе, но название стоит сменить на BuildErrorResponse или что-то в таком духе, поскольку "AlternativeResult" не даёт нам никакой информации

zefirlover committed:
AlternativeResult method changes, pre-dev 5
AlternativeResult was renamed to BuildErrorResponse
Method moved to the end of the TheoryBlockRepository.cs
